confmap/confmap_test.go

```diff
@@ -113,7 +113,6 @@ func TestExpandNilStructPointersHookFunc(t *testing.T) {
 	// assert.False(t, *cfg.Boolean)
 	assert.Nil(t, cfg.Struct)
 	assert.NotNil(t, cfg.MapStruct)
-	// TODO: Investigate this unexpected result.
```

Comment:
This and the other TODO were added in #4880. I don't understand why this is unexpected but in any case I feel like we should just live with it, I don't see a benefit in changing this right now

Reply:
agreed